src/transport_ext.rs

```diff
@@ -31,11 +39,31 @@ pub trait TransportExt: Transport {
     ///
     /// This method returns an `Arc<BandwidthSinks>` that can be used to retreive the total number
     /// of bytes transferred through the sockets.
-    fn with_bandwidth_logging(self) -> (BandwidthLogging<Self>, Arc<BandwidthSinks>)
+    fn with_bandwidth_logging<S>(self) -> (Boxed<(PeerId, StreamMuxerBox)>, Arc<BandwidthSinks>)
```

Comment:
Can you either add a small test, or add a doc example to the `with_bandwidth_logging` method (preferred)?

Reply:
Added a doc test (ignored because it requires certain features)

Reply:
Thanks for adding the test.

In my eyes a doc test that is `ignore`d is not helpful and destined to be outdated. Is there some way to enable the features for testing only? //CC @thomaseizinger

Reply:
Doc tests have access to all `dev-dependencies` so you can simply add whatever you need there and it will be available in the test. You will have to change the imports to though.